superset/dashboards/filter_state/api.py

```diff
@@ -0,0 +1,246 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from typing import Type
+
+from flask import Response
+from flask_appbuilder.api import expose, protect, safe
+
+from superset.dashboards.filter_state.commands.create import CreateFilterStateCommand
+from superset.dashboards.filter_state.commands.delete import DeleteFilterStateCommand
+from superset.dashboards.filter_state.commands.get import GetFilterStateCommand
+from superset.dashboards.filter_state.commands.update import UpdateFilterStateCommand
+from superset.extensions import event_logger
+from superset.key_value.api import KeyValueRestApi
+from superset.views.base_api import statsd_metrics
+
+logger = logging.getLogger(__name__)
+
+
+class DashboardFilterStateRestApi(KeyValueRestApi):
+    class_permission_name = "FilterStateRestApi"
+    resource_name = "dashboard"
+    openapi_spec_tag = "Filter State"
+
+    def get_create_command(self) -> Type[CreateFilterStateCommand]:
+        return CreateFilterStateCommand
+
+    def get_update_command(self) -> Type[UpdateFilterStateCommand]:
+        return UpdateFilterStateCommand
+
+    def get_get_command(self) -> Type[GetFilterStateCommand]:
+        return GetFilterStateCommand
+
+    def get_delete_command(self) -> Type[DeleteFilterStateCommand]:
+        return DeleteFilterStateCommand
```

Comment:
Do you foresee any reasons why these should be methods, and not just class attributes? Eg:

```python
create_command = CreateFilterStateCommand
```

Reply:
It's just to use `ABC` and to make sure children implement the behavior 😉